### ISSUE TITLE
Huge lag in build pane buttons

### ISSUE THREAD

**Issue by david-cortes** [rstudio/rstudio#8387]
In the 1.4 version, there is now a very noticeable lag when clicking any button in the build pane, particularly `Install and Restart`. I find myself clicking it twice quite often as it feels as though it didn't to anything. I recall the lag was not too bad in earlier versions.

**Steps to reproduce:**
* Create some R package project.
* Click `Install and Restart` in the build pane.

**Expected behavior:** should execute the action immediately.

**Actual behavior:** waits some 2-4 seconds before showing anything in the pane.

**Setup:** RStudio 1.4.953, R 4.0.3.

From some experimentation, the lag also seems to depend on the specific package. For example, [rsparse](https://www.github.com/rexyai/rsparse) gets a bigger lag than others,

**Comment by kevinushey** [comment]
I'm not able to reproduce -- even with the rsparse package, output is displayed in the Build pane quickly. Can you provide any other information? Do you by any chance have a `.Rprofile` wherein you do extra work that could slow down the build / install?

**Comment by david-cortes** [comment]
I don't have any custom `.Rprofile`. I've now noticed that the lag seems to occur only after some R usage.

**Comment by ronblum** [comment]
I'm unable to reproduce this—the responses are pretty quick. @david-cortes Is this in Debian Bullseye (11) or Buster (10)? I'm using Buster, so that might be the difference.

**Comment by david-cortes** [comment]
It's in Bullseye, and I'm noticing it tends to happen when I have C++ files opened and have already installed the project package at least once.

**Comment by kevinushey** [comment]
If that's the case, the issue is likely caused by the clang diagnostics system working in the background (and holding up the session) when the build is started.

Does the issue reproduce in other packages which don't include any C++ sources?

**Comment by david-cortes** [comment]
Doesn't reproduce in packages with no C++ code, but I haven't tried many either. There's also a lag in the button to switch projects though.

**Comment by kevinushey** [comment]
Given that this only effects projects using C++ code, I think this is out of scope for 1.4.
